tracee-rules/benchmark/benchmark_test.go

```diff
@@ -0,0 +1,143 @@
+package benchmark
+
+import (
+	"os"
+	"testing"
+	"time"
+
+	"github.com/aquasecurity/tracee/tracee-rules/benchmark/signature/golang"
+	"github.com/aquasecurity/tracee/tracee-rules/benchmark/signature/rego"
+	"github.com/aquasecurity/tracee/tracee-rules/engine"
+	"github.com/aquasecurity/tracee/tracee-rules/types"
+	"github.com/stretchr/testify/require"
+)
+
+const (
+	inputEventsCount = 1000
+)
+
+func BenchmarkOnEventCodeInjectionRegoRule(b *testing.B) {
+	codeInjectSig, err := rego.NewCodeInjectionSignature()
+	require.NoError(b, err)
+
+	err = codeInjectSig.Init(ignoreFinding)
+	require.NoError(b, err)
+
+	for i := 0; i < b.N; i++ {
+		err = codeInjectSig.OnEvent(triggerCodeInjectorPtraceEvent)
+		require.NoError(b, err)
+	}
+}
+
+func BenchmarkOnEventCodeInjectionGoRule(b *testing.B) {
+	codeInjectSig := golang.NewCodeInjectionSignature()
+	err := codeInjectSig.Init(ignoreFinding)
+	require.NoError(b, err)
+
+	for i := 0; i < b.N; i++ {
+		err := codeInjectSig.OnEvent(triggerCodeInjectorPtraceEvent)
+		require.NoError(b, err)
+	}
+}
+
+func BenchmarkEngineWithCodeInjectionRegoRule(b *testing.B) {
+	// Prepare signatures
+	codeInjectSig, err := rego.NewCodeInjectionSignature()
+	require.NoError(b, err)
+
+	sigs := []types.Signature{
+		codeInjectSig,
+	}
+
+	for i := 0; i < b.N; i++ {
+		// Produce events
+		eventsCh := make(chan types.Event, inputEventsCount)
+		ProduceEventsInMemory(eventsCh, inputEventsCount)
```

Comment:
maybe move this out of the bench loop? IMO we can produce the events once

Reply:
How would you Stop then engine (and benchmark iteration) in this approach? We generate 1000 events in each benchmark loop and close done channel when the input events channel is empty.